tests/ISDParsingTests.cpp

```diff
@@ -0,0 +1,315 @@
+}
+
+TEST(ISDParsing, TotalSamples) {
+  json isd = {
+    {"image_samples", 16}
+  };
+  EXPECT_EQ(16, getTotalSamples(isd));
+}
+
+TEST(ISDParsing, StartTime) {
+  json isd = {
+    {"starting_ephemeris_time", -10}
+  };
+  EXPECT_EQ(-10, getStartingTime(isd));
+}
+
+TEST(ISDParsing, CenterTime) {
+  json isd = {
+    {"center_ephemeris_time", -10}
+  };
+  EXPECT_EQ(-10, getCenterTime(isd));
+}
+
+TEST(ISDParsing, IntegrationStartLines) {
+  json isd = {
+    {"line_scan_rate", {
+      {0, 1, 2},
+      {3, 4, 5},
+      {6, 7, 8}}
+    }
+  };
+  std::vector<double> startLines = {0, 3, 6};
+  EXPECT_EQ(startLines, getIntegrationStartLines(isd));
+}
+
+TEST(ISDParsing, IntegrationStartTimes) {
+  json isd = {
+    {"line_scan_rate", {
+      {0, 1, 2},
+      {3, 4, 5},
+      {6, 7, 8}}
+    }
+  };
+  std::vector<double> startTimes = {1, 4, 7};
+  EXPECT_EQ(startTimes, getIntegrationStartTimes(isd));
+}
+
+TEST(ISDParsing, IntegrationTimes) {
+  json isd = {
+    {"line_scan_rate", {
+      {0, 1, 2},
+      {3, 4, 5},
+      {6, 7, 8}}
+    }
+  };
+  std::vector<double> times = {2, 5, 8};
+  EXPECT_EQ(times, getIntegrationTimes(isd));
+}
+
+TEST(ISDParsing, SampleSumming) {
+  json isd = {
+    {"detector_sample_summing", 4}
+  };
+  EXPECT_EQ(4, getSampleSumming(isd));
+}
+
+TEST(ISDParsing, LineSumming) {
+  json isd = {
+    {"detector_line_summing", 4}
+  };
+  EXPECT_EQ(4, getLineSumming(isd));
+}
+
+TEST(ISDParsing, FocalLength) {
+  json isd = {
+    {"focal_length_model", {
+      {"focal_length", 2}}
+    }
+  };
+  EXPECT_EQ(2, getFocalLength(isd));
+}
+
+TEST(ISDParsing, FocalLengthEpsilon) {
+  json isd = {
+    {"focal_length_model", {
+      {"focal_epsilon", 0.1}}
+    }
+  };
+  EXPECT_EQ(0.1, getFocalLengthEpsilon(isd));
+}
+
+TEST(ISDParsing, Focal2PixelLines) {
+  json isd = {
+    {"focal2pixel_lines", {0, 1, 2}}
+  };
+  std::vector<double> coefficients = {0, 1, 2};
+  EXPECT_EQ(coefficients, getFocal2PixelLines(isd));
+}
+
+TEST(ISDParsing, Focal2PixelSamples) {
+  json isd = {
+    {"focal2pixel_samples", {0, 1, 2}}
+  };
+  std::vector<double> coefficients = {0, 1, 2};
+  EXPECT_EQ(coefficients, getFocal2PixelSamples(isd));
+}
+
+TEST(ISDParsing, DetectorCenterLine) {
+  json isd = {
+    {"detector_center", {
+      {"line", 2}}
+    }
+  };
+  EXPECT_EQ(2, getDetectorCenterLine(isd));
+}
+
+TEST(ISDParsing, DetectorCenterSample) {
+  json isd = {
+    {"detector_center", {
+      {"sample", 3}}
+    }
+  };
+  EXPECT_EQ(3, getDetectorCenterSample(isd));
+}
+
+TEST(ISDParsing, DetectorStartingLine) {
+  json isd = {
+    {"starting_detector_line", 1}
+  };
+  EXPECT_EQ(1, getDetectorStartingLine(isd));
+}
+
+TEST(ISDParsing, DetectorStartingSample) {
+  json isd = {
+    {"starting_detector_sample", 2}
+  };
+  EXPECT_EQ(2, getDetectorStartingSample(isd));
+}
+
+TEST(ISDParsing, MinHeight) {
+  json isd = {
+    {"reference_height", {
+      {"minheight", -1},
+      {"unit", "km"}}
+    }
```

Comment:
this tests one of the edge cases I was asking about above, but in, arguably, the wrong place.